src/riscv/linux/hwcap.c

```diff
@@ -0,0 +1,37 @@
+#include <riscv/linux/api.h>
+
+#if CPUINFO_MOCK
+    #include <cpuinfo-mock.h>
+#endif
+
+#if CPUINFO_ARCH_RISCV32 || CPUINFO_ARCH_RISCV64
+    #include <sys/auxv.h>
+#else
+    #define AT_HWCAP 16
+    #define AT_HWCAP2 26
+#endif
+
+#if CPUINFO_MOCK
+    static uint32_t mock_hwcap = 0;
+    void cpuinfo_set_hwcap(uint32_t hwcap) {
+        mock_hwcap = hwcap;
+    }
+
+	static uint32_t mock_hwcap2 = 0;
+	void cpuinfo_set_hwcap2(uint32_t hwcap2) {
+		mock_hwcap2 = hwcap2;
+	}
+#endif
+
+#if CPUINFO_ARCH_RISCV32 || CPUINFO_ARCH_RISCV64
```

Comment:
Why is this guarded with ARCH type? BTW I started adding tests for the visionfive board recently: #200 - does this satisfy the hwcap / riscv_hwprobe mocking bits?